pkg/templates/templates.go

```diff
@@ -479,20 +480,28 @@ func (t *TemplateResolver) processForDataTypes(str string) string {
 	// outer quotes around key-values are always single quotes
 	// even if the user input is with  double quotes , the yaml processed and saved with single quotes
 
-	d1 := regexp.QuoteMeta(t.config.StartDelim)
-	d2 := regexp.QuoteMeta(t.config.StopDelim)
-	re := regexp.MustCompile(
-		`:\s+(?:[\|>]-?\s+)?(?:'?\s*)(` + d1 + `.*\|\s*(?:toInt|toBool|toLiteral).*` + d2 + `)(?:\s*'?)`,
-	)
-	klog.V(2).Infof("\n Pattern: %v\n", re.String())
+	if strings.Contains(str, "toInt") ||
```

Comment:
I think you misunderstood my previous comment. This still requires scanning the entire string up to 5 times but yet the regex does this parsing for us anyways.

I think you could just remove these if statements and then return early if the length of `submatchall` is `0`.

Reply:
got it.  I have made another pass at addressing this.